homeassistant/components/nasweb/switch.py

```diff
@@ -0,0 +1,136 @@
+"""Platform for NASweb output."""
+
+from __future__ import annotations
+
+import logging
+import time
+from typing import Any
+
+from webio_api import Output as NASwebOutput
+
+from homeassistant.components.switch import DOMAIN as DOMAIN_SWITCH, SwitchEntity
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers.device_registry import DeviceInfo
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+import homeassistant.helpers.entity_registry as er
+from homeassistant.helpers.typing import DiscoveryInfoType
+from homeassistant.helpers.update_coordinator import (
+    BaseCoordinatorEntity,
+    BaseDataUpdateCoordinatorProtocol,
+)
+
+from . import NASwebConfigEntry
+from .const import DOMAIN, STATUS_UPDATE_MAX_TIME_INTERVAL
+from .coordinator import NASwebCoordinator
+
+OUTPUT_TRANSLATION_KEY = "switch_output"
+
+_LOGGER = logging.getLogger(__name__)
+
+
+def _get_output(coordinator: NASwebCoordinator, index: int) -> NASwebOutput | None:
+    for out in coordinator.webio_api.outputs:
+        if out.index == index:
+            return out
+    return None
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config: NASwebConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+    discovery_info: DiscoveryInfoType | None = None,
+) -> None:
+    """Set up switch platform."""
+    coordinator = config.runtime_data
+    current_outputs: set[int] = set()
+
+    @callback
+    def _check_entities() -> None:
+        nonlocal current_outputs
```

Comment:
I don't think `nonlocal` is needed here, since we just mutate `current_outputs`, but don't initialize?
```suggestion
```